new-chatbot-app/src/llm-chain/llm-toolbox/libcal-tools/check-open-hour-tool/check-open-hour-tool.service.ts

```diff
@@ -0,0 +1,183 @@
+    return `${year}-${month}-${day}`;
+  }
+
+  private getMondayAndSundayOfWeek(dateStr: string): string[] {
+    const dateParts = dateStr.split('-');
+    const year = parseInt(dateParts[0]);
+    const month = parseInt(dateParts[1]) - 1; // Month is zero-based (0 = January, 1 = February, etc.)
+    const day = parseInt(dateParts[2]);
+
+    const date = new Date(year, month, day);
+    const currentDay = date.getDay(); // 0 = Sunday, 1 = Monday, ..., 6 = Saturday
+
+    // Calculate the difference in days between the input date and Monday
+    const daysToMonday = (currentDay + 6) % 7;
+
+    // Calculate the date of Monday
+    const monday = new Date(year, month, day - daysToMonday);
+
+    // Create an array of date strings from Monday to Sunday
+    const datesOfWeek: string[] = [];
+    for (let i = 0; i < 7; i++) {
+      const newDate = new Date(
+        monday.getFullYear(),
+        monday.getMonth(),
+        monday.getDate() + i,
+      );
+      datesOfWeek.push(this.formatDateString(newDate));
+    }
+    return datesOfWeek;
+  }
+
+  public async toolRunForLlm(toolInput: {
+    date: string | null | undefined;
+  }): Promise<string> {
+    if (
+      toolInput.date === null ||
+      toolInput.date === 'null' ||
+      toolInput.date === undefined ||
+      toolInput.date === 'undefined' ||
+      toolInput.date === ''
+    ) {
+      return 'Cannot check the building hour without a date. Ask the customer to provide the date before checking.\n';
+    }
+
+    try {
+      const response = `Open Hour of the requested week:
+      ${JSON.stringify(
+        await this.run(toolInput.date),
+      )}.\nIf any day does not exist in the array, the library does not open that day. Always answer with both open hour and close hour to the customer.\n`;
+      return response;
+    } catch (error: any) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  /**
+   * Run the check open hour service.
+   * @param date
+   * @returns
+   */
+  async run(date: string): Promise<WeekAvailability> {
+    const header = {
+      Authorization: `Bearer ${this.accessToken}`,
+    };
+    const weekdays = this.getMondayAndSundayOfWeek(date);
+
+    try {
+      const dayNames: WeekDay[] = [
+        'monday',
+        'tuesday',
+        'wednesday',
+        'thursday',
+        'friday',
+        'saturday',
+        'sunday',
+      ];
+
+      const response = await this.httpService.axiosRef.get(
+        `${this.OPEN_HOUR_URL}/8113`,
+        {
+          params: {
+            from: weekdays[0],
+            to: weekdays[weekdays.length - 1],
+          },
+          headers: header,
+        },
+      );
+
+      let filteredData = weekdays.reduce(
+        (prevObj, currentDay, currentIndex) => {
+          return {
+            ...prevObj,
+            [dayNames[currentIndex]]: response.data[0].dates[currentDay].hours,
+          };
+        },
+        {},
+      );
```

Comment:
weekdays have 7 elements and the response.data[0].dates only have one date '2024-04-09' based on your test input on line 90 in file .spec

Reply:
Yeah I know but what I mean is the data returned from http query only has 1 element instead of 7. You can see that I added params to config for the query but it still returns 1 only

Reply:
http query supposes to return 7 days. Look at the line 148, 149. We explicitly request it to return 7 days

Reply:
I requested for 6-7 days and it returns 6-7 elements
![image](https://github.com/Meng-V/smartchatbot/assets/62295931/a0770c81-726f-4202-81af-f8bc57fafccc)